### PR TITLE
Fix null pointers check

### PR DESCRIPTION
**Pull request by gromchek**
Added checks to few pointers

### CODE REVIEW

File: src/ai/script_ai.cpp [hunk 1]
```diff
@@ -1625,11 +1625,12 @@ static int CclDefineAiPlayer(lua_State *l)
 		if (value == "ai-type") {
 			const std::string_view aiName = LuaToString(l, j + 1);
 			CAiType *ait = GetAiTypesByName(aiName);
-			if (ait == nullptr) {
+			if (ait) {
+				ai.AiType = ait;
+				ai.Script = ait->Script;
+			} else {
 				LuaError(l, "ai-type not found: %s", aiName.data());
```
Comment (Jarod42) [inline src/ai/script_ai.cpp:1632]: `LuaError` should not `return`, so reordering code is not required.
Same for src/unit/depend.cpp

Comment (gromchek) [reply src/ai/script_ai.cpp:1632]: Can you please describe what do you mean by "LuaError should not return"?

Comment (Jarod42) [reply src/ai/script_ai.cpp:1632]: As `abort`/`exit`, `LuaError` stops the program.

Comment (gromchek) [reply src/ai/script_ai.cpp]: Thanks. I thought about that but decided to clarify.

**Pull request merged**